sdlib/d/gc/tcache.d

```diff
@@ -54,6 +64,77 @@ public:
 		pd.arena.free(emap, pd, ptr);
 	}
 
+	/**
+	 * Appendables API.
+	 */
+
+	// Get the capacity of the array slice denoted by slice and length.
+	// It includes the space occupied by the slice itself.
+	size_t getArrayCapacity(void* sliceAddr, size_t sliceLen) {
+		if ((sliceAddr is null) || (!sliceLen))
+			return 0;
+
+		auto pd = maybeGetPageDescriptor(sliceAddr);
+		// Return zero if slice is unknown to GC or is to a non-appendable block:
+		if ((pd.extent is null) || (!pd.extent.allocSize))
```

Comment:
You don't need to check if the alloc size is zero, as the next check will fail anyway.